### ISSUE TITLE
Ground Truth noise level problem

### ISSUE THREAD

**Issue by WXR1998** [IDKiro/CBDNet-tensorflow#8]
In the git repo of the original author of the paper, he said that the noise level is defined by:
X*sigma_s + sigma_c, where X indicates the clean picture. So the ground truth of noise level map should be built in this way. But in your code, the GT is calculated by simply **noisy_pic - clean_pic**, which does not exactly square with the original paper.

**Comment by IDKiro** [comment]
Img_{noisy} = Img_{clean} + noise.

I have calculated the noise using X*sigma_s + sigma_c before and add it to clean images to get noisy images. 
Why can't I get the added noise in this way?

And ,can you tell me how to get the noise using X*sigma_s + sigma_c when it is real-world noisy image (such as image pairs in RENOIR).

**Comment by WXR1998** [comment]
In my opinion, the noise level should be a non-negative number. It indicates the strength of the noise. But in the synthetic picture generating process, a zero-mean noise is added to the picture, that means the noise value may be negative. If simply perform a subtraction between these two pictures, it may cause some problem.

**Comment by WXR1998** [comment]
And by the way, the last layer of FCN utilizes ReLU as its activate function, that means the output of this layer cannot be negative. In this way, the negative values of GT noise level map may not be fitted, I think.

**Comment by IDKiro** [comment]
What you said is that i consider before, but i don't understand how to get the noise levels of real-world noisy images. 
At the time of the code was written, I thought that the sub-network will only fit the positive value because the negative value contribution same to loss for the same input image pair.
Maybe it is because what i read is the v1 paper, I just saw that the paper has been updated.
In v1 paper, the author does not discuss how to get sigma(y) but just how to calculate the noise model.
Your question is very meaningful. I suggest you read the official code. After I wrote this code, they added the python version of the noise model code.
I am very sorry that I have no time to re-read the new version paper. I am glad that if you can open a pull request after you know how to get true noise level maps.

**Comment by IDKiro** [comment]
Now I remember that it don’t need ground truth noise level maps when the network was trained on real noisy images (even I actually have distinguished between the two cases in the code). It is my miss, i have understand how to fix it.

**Comment by WXR1998** [comment]
I have got the idea of calculating the noise level from the issue column in the author's repo, too :). 
And in the version of the paper I read, the author didn't mention it, either.
Thank you for your patience!
